api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -275,6 +275,13 @@ message ExternalProcessor {
   // backend stream lifetime. In this case, Envoy will eventually timeout the external processor stream according to this time limit.
   // The default value is 5000 milliseconds (5 seconds) if not specified.
   google.protobuf.Duration deferred_close_timeout = 19;
+
+  // [#not-implemented-hide:]
+  // Enables sending body to the side stream server before receives the header response from that server.
+  // It only works for STREAMED body processing mode. For any other body processing modes, it is ignored.
```

Comment:
High-level question: Is this not going to be useful for BUFFERED use-cases that don't require sending a headers response?
Just in case this may be needed in the future, I think removing the word "streaming" from the field name will make sense.

Reply:
I think your question is that we might need to do this for all body modes.  That's a good question.

For BUFFERED mode, currently the headers are already held in Envoy before the body response coming back.   So, technically, no need to do this for BUFFERED mode.

Reply:
Interesting... I can think of a scenario where the headers are sent to the side-stream, the first piece of body of the request arrives at the Envoy and it is then buffered, and then the second (and last) piece of body arrives at the Envoy, and it may want to send it then to the side stream (before receiving the headers).
But I guess this is moot now, that the API has changed considerably.

Reply:
Sorry, let me clarify the BUFFERED mode a little bit:

Header arrives Envoy -----> Send to the side stream server -----> Envoy receives the header response ----> Keep it in Envoy (i.e, not forwarding to the upstream).

Body arrives Envoy -----> Envoy Buffer all the data, until end-of-stream received  (if the data is too large, exceeding the watermark then it trigger an error) ---> Send to the side stream server. -----> Envoy receives the body response (this response may or may not contain header mutation as well) -----> Envoy sends both header and body to the upstream server.

Reply:
The key in BUFFERED mode is that the header is not forwarded to the backend server until the body response comes back to Envoy. And the side stream server can mutate the header again after it sees the body. That gives the side stream a chance to change routing based on body content.